src/Control/Carrier/Choose/Church.hs

```diff
@@ -20,11 +21,15 @@ import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
 import Data.Maybe (fromJust)
+import qualified Data.Semigroup as S
 import Prelude hiding (fail)
 
 runChoose :: (m b -> m b -> m b) -> (a -> m b) -> ChooseC m a -> m b
 runChoose fork leaf m = runChooseC m fork leaf
 
+runChooseS :: (Applicative m, S.Semigroup b) => (a -> m b) -> ChooseC m a -> m b
```

Comment:
Can we get documentation for this, mentioning that it uses `<>` to merge its results?